test/integration/targets/fetch/tasks/main.yml

```diff
@@ -139,3 +139,77 @@
     that:
       - "failed_fetch_dest_dir is failed"
       - "failed_fetch_dest_dir.msg"
+
+# Set up for some 'become' tests. Like the 'copy' module tests
+# we first need to have an unprivileged user with the same 
+# ssh key as our regular user allowed for login.
+- name: note the home directory for the Ansible user
+  set_fact: 
+    remote_user_home: "{{ ansible_env.HOME }}"
+
+- name: create a file that we can use to fetch using become
+  become: true
+  copy: content="test" owner=root mode=0600 dest={{ remote_tmp_dir }}/orig.root
+
+- name: Create remote unprivileged remote user
+  become: true
+  user:
+    name: '{{ remote_unprivileged_user }}'
+  register: user
+
+- name: create an `.ssh` directory for holding `authorized_keys`
+  become: true
+  file:
+    path: "{{ user.home }}/.ssh"
+    owner: '{{ remote_unprivileged_user }}'
+    state: directory
+    mode: 0700
+
+- name: Duplicate authorized_keys
+  become: true
+  copy:
+    src: '{{ remote_user_home }}/.ssh/authorized_keys'
+    dest: '{{ user.home }}/.ssh/authorized_keys'
+    owner: '{{ remote_unprivileged_user }}'
+    mode: 0600
+    remote_src: yes
+
+# Pulling a root-only file as a non-root user should fail, but it succeeds.
+# This appears to be related to the closed issue #13388.
+#
+# Based on https://github.com/ansible/ansible/pull/13424 it appears that overriding
+# remote_user on a per-task basis with Docker doesn't work. That matches the observed
+#   <testhost> ESTABLISH LOCAL CONNECTION FOR USER: root
+# entries in -vvv output.
+#
+# https://github.com/ansible/ansible/pull/15556 had a good conversation about the
+# Docker-specific challenges of the "become" process.
+- name: attempt to fetch the only-root-readable test file using a non-root user, should fail
+  fetch: src={{ remote_tmp_dir }}/orig.root dest={{ output_dir }}/fetched
+  remote_user: '{{ remote_unprivileged_user }}'
```

Comment:
Tests are run with `ansible_connection=local`. The local connection plugin ignores `remote_user`, so this will have no effect.

Reply:
Thanks, @mattclay. What would be a good way to test this?

In #13388 there was some mention of applying a different user to the `docker exec` commands, and in pull request #13424 it looks like it was implemented. However, as you mentioned, when using `local.py` none of that matters since `docker.py` won't even be used.

If `remote_user` gets ignored, that definitely makes it a challenge to test. Any thoughts?

Reply:
That sounds like something that should be added to the docker_connection integration test.

You could rename the existing `runme.sh` for that test to `test.sh` and create a new `runme.sh` that invokes the original tests as well as the new ones you'd need to add.